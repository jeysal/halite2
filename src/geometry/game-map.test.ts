import GameMap from 'halite/GameMap';

import {
  centerOfMap,
  cornersOfMap,
  diameter,
  distanceFromCenterOfMap,
} from './game-map';

describe('centerOfMap', () => {
  test('calculates the center of the map', () => {
    expect(
      centerOfMap(new GameMap({ myPlayerId: 0, width: 100, height: 100 })),
    ).toEqual({ x: 50, y: 50 });
  });
});

describe('distanceFromCenter', () => {
  test('calculates the distance from the center of the map', () => {
    expect(
      distanceFromCenterOfMap(
        new GameMap({ myPlayerId: 0, width: 100, height: 100 }),
        { x: 53, y: 54 },
      ),
    ).toBe(5);
  });
});

describe('cornersOfMap', () => {
  test('returns the corners of the map', () => {
    expect(
      cornersOfMap(new GameMap({ myPlayerId: 0, width: 100, height: 100 })),
    ).toMatchSnapshot();
  });
});

describe('diameter', () => {
  test('calculates the diameter of the map', () => {
    expect(
      diameter(new GameMap({ myPlayerId: 0, width: 30, height: 40 })),
    ).toBe(50);
  });
});
