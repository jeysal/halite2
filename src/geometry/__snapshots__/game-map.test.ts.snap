// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cornersOfMap returns the corners of the map 1`] = `
Array [
  Object {
    "x": 0,
    "y": 0,
  },
  Object {
    "x": 0,
    "y": 100,
  },
  Object {
    "x": 100,
    "y": 0,
  },
  Object {
    "x": 100,
    "y": 100,
  },
]
`;
